### PR TITLE
journal: Handle partially read HashItem's when punching holes

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: src/journal/journald-file.c [hunk 1]
```diff
@@ -97,6 +97,9 @@ static int journald_file_punch_holes(JournalFile *f) {
                 if (n < 0)
                         return n;
 
+                /* Let's ignore any partial hash items by rounding down to the nearest multiple of HashItem. */
+                n -= n % sizeof(HashItem);
```
Comment (keszybz) [inline src/journal/journald-file.c:101]: I don't understand. Isn't the loop below enough to achieve the same effect? `(size_t) n / sizeof(HashItem)` rounds down too…

**Review by yuwata — APPROVED** [review]
LGTM.

File: src/journal/journald-file.c (hunk 1 above)
Comment (yuwata) [reply src/journal/journald-file.c:101]: If `n` is not a multiple of `sizeof(HashItem)`, then the offset of next `pread()` becomes strange.

Comment (keszybz) [reply src/journal/journald-file.c:101]: Hmm, doesn't this increase the risk that we end up in an infinite loop here? If the file is strangely truncated, pread() could return something between 0 and sizeof(HashItem). Shouldn't the loop above if `n < 0` or the rounded-down `n` is 0?

Comment (yuwata) [reply src/journal/journald-file.c:101]: Indeed.

Comment (daandemeyer) [reply src/journal/journald-file.c:101]: Added a n > 0 check to the outer loop.

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**